### PR TITLE
Add release notes

### PR DESCRIPTION
**Pull request by akien-mga**
I've only written the release notes / end user-friendly changelog for OD 0.4.9 so far, based on #9. Feel free to review and comment on the form while I prepare the 0.5.0 release notes.

You can check the file directly in my fork on GitHub to see the Markdown formatting live: https://github.com/akien-mga/OpenDungeons/blob/development/RELEASE-NOTES.md

### CODE REVIEW

**Comment by akien-mga** [comment]
Before making the 0.5.0 release notes I'll try to make sure that #418 has an accurate listing of what we did.

For the release notes I removed most lines related to fixing bugs in the new features: end users only want to know that there is a new feature, and what is it's status, not how it was in the development branch :)

**Comment by akien-mga** [comment]
I added the release notes for 0.5.0. Feel free to comment on those (maybe the simplest would be to merge and to make another pull request if you want to do many adjustments).

**Comment by akien-mga** [comment]
There were quite a few awesome features (like claimable portals and portal waves) missing from #418, so I added them in the release notes and also added some stuff to #418, but maybe not everything.
There is so much stuff in this new release, it's hard to keep track of everything. Yet another reason for releasing more often, but guys, you implement awesome stuff too fast :-P

IMO for the release announcement on the website and the forum, we should give a link to this RELEASE-NOTES.md on Github instead of #418, it will be more user-focused and maybe more accurate.

**Comment by Bertram25** [comment]
> IMO for the release announcement on the website and the forum, we should give a link to this RELEASE-NOTES.md on Github instead of #418, it will be more user-focused and maybe more accurate.

Yeah, sure.

Excellent work, btw. :D

File: RELEASE-NOTES.md [hunk 1]
```diff
@@ -0,0 +1,209 @@
...
+  - Beds are removed when creatures die.
+* Hatchery & food system (#13)
+  - All models integrated and the room can placed in game (#106)
+  - The room spawns chickens wandering within the room limits.
+  - Creatures can eat them visually and calm their hunger.
+* Forge (not actually useful in this release)
+* Library (not actually useful in this release)
+* Crypt: spawns gnomes when you let enemy creatures decay there (#322)
+* Add support to destroy a room using a new button. Destroying a room gives back half the tiles price (#108)
+* Add support to disable a room for a specific player in a specific level (#86)
```
Comment (Bertram25) [inline RELEASE-NOTES.md]: This was done for 0.5.0, IMHO, through researchable/unresearchable rooms.

File: RELEASE-NOTES.md [hunk 2]
```diff
@@ -0,0 +1,209 @@
...
+
+**Gameplay:**
+
+* Implemented the Fog of War (#295)
+  - Implemented trap vision logic (#408)
+* Implemented creatures pay day (#431)
+* Implemented spells support and made workers spawned only via a spell (#16)
+* Implemented the Call to War spell (#484)
+* Implemented the library's research logic + research tree (#411 #508)
+* Portals can be claimabled (#588)
```
Comment (Bertram25) [inline RELEASE-NOTES.md]: Claimed, I guess.

File: RELEASE-NOTES.md (hunk 1 above)
Comment (hwoarangmy) [inline RELEASE-NOTES.md]: I confirm

**Comment by Bertram25** [comment]
My comment apart, this can be merged.

File: RELEASE-NOTES.md (hunk 1 above)
Comment (akien-mga) [inline RELEASE-NOTES.md]: Indeed, looks like #9 was not up to date. I'll fix that.

**Comment by akien-mga** [comment]
I made the requested changes :)

File: RELEASE-NOTES.md [hunk 3]
```diff
@@ -0,0 +1,209 @@
...
+  - Beds are persistent on room expansion.
+  - Beds are removed when creatures die.
+* Hatchery & food system (#13)
+  - All models integrated and the room can placed in game (#106)
+  - The room spawns chickens wandering within the room limits.
+  - Creatures can eat them visually and calm their hunger.
+* Forge (not actually useful in this release)
+* Library (not actually useful in this release)
+* Crypt: spawns gnomes when you let enemy creatures decay there (#322)
+* Add support to destroy a room using a new button. Destroying a room gives back half the tiles price (#108)
```
Comment (hwoarangmy) [inline RELEASE-NOTES.md]: Traps can also be destroyed like rooms

Comment (akien-mga) [inline RELEASE-NOTES.md]: True, I'll fix that.

**Comment by hwoarangmy** [comment]
Seems good to me. Merging

**Comment by akien-mga** [comment]
What would be the best method to backport this PR to the 0.5 branch?

**Comment by Bertram25** [comment]
git cherry-pick
IMHO. I'll do it if nobody else does.

**Comment by akien-mga** [comment]
Should one run git cherry-pick on all commits, or is there a way to cherry pick a whole PR?

**Comment by Bertram25** [comment]
> Should one run git cherry-pick on all commits, or is there a way to cherry pick a whole PR?

There are several ways. You can actually use cherry-pick with a commit range, even if that's not very practical usually:
`git cherry-pick hash1..hash2` for instance.

you can also use the rebase --onto method, but you'd better need actual branches for that:
rebase every commit after 'b' and put it "onto" aka on the top of 'a'
`git rebase --onto a b`
a,b can be branch names or commit hashes.

The spirit behind the two, in any case, (and it is what I usually do) is to have the commits to cherry-pick in a separate branch and pull that branch into what you want, even a branch from a personal fork.

For instance if you did the PR, you can keep the branch in your personal fork with the commits that were merged, and use that to be pull on stable afterwards:
(From the main repo)
`git pull --rebase git://github.com/akien-mga/OpenDungeons.git branch-with-interesting-changes`

 If things are more twisted, it is usually a better thing to take commits one by one, ending in a precise cherry-picking.

I hope all this helped. Of course, if anyone has got better way to handle that, I'm all ears. :) git is always evolving after all.

**Comment by akien-mga** [comment]
Thanks for the howto, I'll come back to it next time I need to backport a lot of stuff. Currently my fork only contains stuff relevant for 0.5.0, so I guess it can be used to update the 0.5 branch once my new PR is merged.

**Pull request merged**